core/blocktree/node.go

```diff
@@ -24,37 +24,38 @@ import (
 	"github.com/disiqueira/gotree"
 )
 
-// node is an element in the BlockTree
-type node struct {
-	hash     common.Hash // Block hash
-	parent   *node       // Parent node
-	number   *big.Int    // Block number
-	children []*node     // Nodes of children blocks
-	depth    *big.Int    // Depth within the tree
+// Node is an element in the BlockTree
+type Node struct {
```

Comment:
I'm not convinced this should exported. IMO any retrieval from the blocktree in another package should return a block. This reduces the need to store duplicate data in the tree and seems to make more sense conceptually as a `node` doesn't really make sense outside the context of the blocktree. Lets chat about this